applications/StructuralMechanicsApplication/python_scripts/element_sensitivity_domain_integration_process.py

```diff
@@ -0,0 +1,109 @@
+        settings -- Kratos parameters containing solver settings.
+        """
+        KratosMultiphysics.Process.__init__(self)
+
+        ## Settings string in json format
+        default_parameters = KratosMultiphysics.Parameters("""{
+            "help"                             : "This class integrates element sensitivities within domains defined by sub model parts",
+            "element_sensitivity_variables"    : [],
+            "model_part_name"                  : "",
+            "sensitivity_model_part_name"      : "",
+            "sensitivity_sub_model_part_list"  : []
+        }""")
+
+        ## Overwrite the default settings with user-provided parameters
+        params.ValidateAndAssignDefaults(default_parameters)
+        self.params = params
+        self.model  = model
+
+    def ExecuteInitialize(self):
+        """ This method is executed at the begining to initialize the process
+
+        Keyword arguments:
+        self -- It signifies an instance of a class.
+        """
+
+        model_part_name = self.params["model_part_name"].GetString()
+        sensitivity_model_part_name = self.params["sensitivity_model_part_name"].GetString()
+        sensitivity_sub_model_part_names = [ self.params["sensitivity_sub_model_part_list"][i].GetString() for i in range( 0,self.params["sensitivity_sub_model_part_list"].size() ) ]
+
+        # Get sensitivity model part
+        if (sensitivity_model_part_name != ""):
+            self.sensitivity_model_part = self.model[model_part_name].GetSubModelPart(sensitivity_model_part_name)
+        else:
+            self.sensitivity_model_part = self.model[model_part_name]
+
+        # Get defined sub model parts of sensitivity model part as integration domains
+        self.sensitivity_sub_model_parts = []
+        if len(sensitivity_sub_model_part_names) is 0:
+            self.sensitivity_sub_model_parts.append(self.sensitivity_model_part)
+        else:
+            for mp_name in sensitivity_sub_model_part_names:
+                self.sensitivity_sub_model_parts.append(self.sensitivity_model_part.GetSubModelPart(mp_name))
+
+        self.element_sensitivity_variables = self.__generate_variable_list_from_input(self.params["element_sensitivity_variables"])
+
+    def ExecuteFinalizeSolutionStep(self):
+        """ This method is executed in order to finalize the current step
+
+        Here the dictionary containing the solution is filled
+
+        Keyword arguments:
+        self -- It signifies an instance of a class.
+        """
+        # loop over integration domains
+        for sub_mp_i in self.sensitivity_sub_model_parts:
+            # loop over sensitivty variables for which integration should performed
+            for variable_i in self.element_sensitivity_variables:
+                value = 0.0
+                # loop over elements in sensitivity sub model parts
+                if next(sub_mp_i.Elements.__iter__() ).Has(variable_i):
```

Comment:
I think this is python3-only
Not sure if it is relevant but [here](https://github.com/KratosMultiphysics/Kratos/blob/master/applications/FluidDynamicsApplication/python_scripts/fluid_solver.py#L179-L183) is how it is solved in the fluid-solver